examples/air_to_async/channel_ops/2mm/mmult.mlir

```diff
@@ -0,0 +1,123 @@
+module attributes {torch.debug_module_name = "model"} {
+  memref.global "private" @channel_7 : memref<i64> = dense<0>
+  memref.global "private" @channel_6 : memref<i64> = dense<0>
+  memref.global "private" @channel_5 : memref<i64> = dense<0>
+  memref.global "private" @channel_4 : memref<i64> = dense<0>
+  memref.global "private" @channel_3 : memref<i64> = dense<0>
+  memref.global "private" @channel_2 : memref<i64> = dense<0>
+  memref.global "private" @channel_1 : memref<i64> = dense<0>
+  memref.global "private" @channel_0 : memref<i64> = dense<0>
+  func.func @forward(%arg0: memref<32x32xi32>, %arg1: memref<32x32xi32>, %arg2: memref<32x32xi32>, %arg3: memref<32x32xi32>) attributes {llvm.emit_c_interface} {
+    %c0 = arith.constant 0 : index
+    %c32 = arith.constant 32 : index
+    %c1 = arith.constant 1 : index
+    %c0_i32 = arith.constant 0 : i32
+    %alloc = memref.alloc() {alignment = 64 : i64} : memref<32x32xi32>
+    linalg.fill ins(%c0_i32 : i32) outs(%alloc : memref<32x32xi32>)
+    %alloc_0 = memref.alloc() {alignment = 64 : i64} : memref<32x32xi32>
+    memref.copy %alloc, %alloc_0 : memref<32x32xi32> to memref<32x32xi32>
+    %0 = memref.get_global @channel_0 : memref<i64>
+    %1 = builtin.unrealized_conversion_cast %0 : memref<i64> to memref<i64>
```

Comment:
The PR looks great to me. Are those unrealized conversion casts generated from -air-to-std? If so, in one recent PR the -air-to-std pass has been extended to support channel puts and gets, without generating those ops.

Reply:
I agree with Steve's point that it would be helpful to add regression tests which test those features, so as to facilitate future code development on top of your work.

Reply:
The unrealized conversion casts are generated by `-air-to-async`. I have moved tests from examples to `mlir/tests`